extensions/parquet/base/src/main/java/io/deephaven/parquet/base/ParquetFileReader.java

```diff
@@ -236,8 +236,16 @@ private static void buildChildren(Types.GroupBuilder builder, Iterator<SchemaEle
             }
 
             if (schemaElement.isSetLogicalType()) {
-                ((Types.Builder) childBuilder)
-                        .as(getLogicalTypeAnnotation(schemaElement.logicalType));
+                LogicalType logicalType = schemaElement.logicalType;
+                if (logicalType.isSetTIMESTAMP()) {
+                    TimestampType timestamp = logicalType.getTIMESTAMP();
+                    if (!timestamp.isAdjustedToUTC) {
```

Comment:
Is it possible to not be adjusted, but to still be in UTC?

Reply:
**Short answer:** A timestamp that is not adjusted is assumed to be in local timezone, irrespective of what the timezone is. So yes, if local timezone is UTC, then this would be in UTC.

**Long answer:** Let me try to summarize what the [official doc](https://github.com/apache/parquet-format/blob/master/LogicalTypes.md) says:
-  A TIMESTAMP with `isAdjustedToUTC=true` is defined as the time elapsed since the Unix epoch. So we can translate such a instant to UTC time and then  adjust it to local timezone. For example, TIMESTAMP(isAdjustedToUTC=true, unit=MILLIS) value of 172800000 corresponds to 1970-01-03 00:00:00 UTC and will change according to timezone.

- A TIMESTAMP with `isAdjustedToUTC=false` represents time fields in a local timezone, regardless of what local timezone is. For example, TIMESTAMP(isAdjustedToUTC=false, unit=MILLIS) value of 172800000 corresponds to 1970-01-03 00:00:00 in any timezone.

Coming back to our issue, when `isAdjustedToUTC=false`, our code crashes with a random exception in some cases and assumes it to be in UTC in other cases. So I added a clear exception for both cases. There is another issue #976 for properly handling non-UTC adjusted timestamps.

Reply:
The point of my question was:
If we see `isAdjustedToUTC=false` and the timezone is specified as `UTC`, can we just treat that as an `Instant`? Is there a timezone in the type metadata?

Reply:
My bad. So no, there is no timezone in the TIMESTAMP type, we just have two fields : isAdjustedToUTC (= true/false)
unit (= MILLIS/MICROS/NANOS). So that information combined with the local timezone gives us how to interpret the timestamp value.

Reply:
Oh. That's horrifying. "This data means something different if you read it in Hong Kong than if you read it in New York."